### PR TITLE
[feature] support for BERT fp16 inference

### PR DESCRIPTION
**Pull request by eric-haibin-lin**
## Description ##
I changed some transformer blocks to support fp16. Specifically:
- input token ids remains fp32
- adjust -inf mask value for softmax
- use fp32 for layernorm

Inference time on p3.2xlarge:
- fp32: 110.98 sec
- fp16: 58.61 sec

Inference accuracy with pre-trained fp32 model:
- fp32: 'exact_match': 81.21097445600756, 'f1': 88.4551346176558
- fp16:  'exact_match': 81.17313150425733, 'f1': 88.43968274854582

@zhiheng-huang

TODO:
Ideally the LayerNorm operator in MXNet should supports fp16 data with stable reduction for std calculation. 

https://github.com/apache/incubator-mxnet/issues/14073
https://github.com/apache/incubator-mxnet/issues/14072

### Essentials ###
- [ ] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by eric-haibin-lin** [comment]
fp16 added in #505. The static QA script will be moved to master branch by @cgraywang 
Still waiting for fp16 support for layer norm and norm in MXNet.
